src/index.js

```diff
@@ -48,12 +48,18 @@ const InputNumber = React.createClass({
     this.componentDidUpdate();
   },
 
+  componentWillUpdate() {
+    this.start = this.refs.input.selectionStart;
+    this.end = this.refs.input.selectionEnd;
+  },
+
   componentDidUpdate() {
     if (this.props.focusOnUpDown &&
       this.state.focused &&
       document.activeElement !== this.refs.input) {
       this.focus();
     }
+    this.refs.input.setSelectionRange(this.start, this.end);
```

Comment:
Do i need to worry about  `setSelectionRange` api compatibility in mobile browser ?
since this component is used for antd-mobile too.